graspologic/nominate/VNviaSGM.py

```diff
@@ -0,0 +1,387 @@
+        subgraph_AA = _get_induced_subgraph_list(
+            AA, self.order_voi_subgraph, voi_reord, mindist=1
+        )
+
+        close_seeds = np.intersect1d(subgraph_AA, seeds_reord)
+
+        if len(close_seeds) <= 0:
+            warnings.warn(
+                "Voi {} was not a member of the induced subgraph A[{}], \
+                Try increasing order_voi_subgraph".format(
+                    voi, seedsA
+                )
+            )
+            self.n_seeds_ = None
+            self.nomination_list_ = None
+            return self
+
+        # Generate the two induced subgraphs that will be used by the matching
+        # algorithm using the seeds that we identified in the previous step.
+        verts_A = _get_induced_subgraph_list(
+            AA, self.order_seeds_subgraph, close_seeds, mindist=0
+        )
+
+        verts_B = _get_induced_subgraph_list(
+            BB, self.order_seeds_subgraph, close_seeds, mindist=0
+        )
+
+        # Determine the final reordering for the graphs that include only
+        # the vertices found by the induced subgraphs in the previous step
+        # For graph A, its of the form (close_seeds, voi, rest in verts_A
+        # in num order). For graph B its of the form (close_seeds, rest in
+        # verts_B in num order)
+        paramed_verts = np.append(close_seeds, voi_reord)
+        ind1 = np.append(
+            np.append(close_seeds, voi_reord), np.setdiff1d(verts_A, paramed_verts)
+        )
+        ind2 = np.concatenate((close_seeds, np.setdiff1d(verts_B, close_seeds)))
+
+        # Generate adjacency matrices for the ordering found in the prev step
+        AA_fin = AA[ind1][:, ind1]
+        BB_fin = BB[ind2][:, ind2]
+
+        # Record the number of seeds used because this may differ from the number
+        # of seeds passed. See the step where close_seeds was computed for an
+        # explanation
+        self.n_seeds_ = len(close_seeds)
+        seeds_fin = np.arange(self.n_seeds_)
+
+        # Call the SGM algorithm using user set parametrs and generate a prob
+        # vector for the voi.
+        sgm = GMP(
+            **self.graph_match_kws,
+        )
+
+        prob_vector = np.zeros((max(AA_fin.shape[0], BB_fin.shape[0]) - self.n_seeds_))
+
+        for ii in range(self.n_init):
+            sgm.fit(AA_fin, BB_fin, seeds_A=seeds_fin, seeds_B=seeds_fin)
+            prob_vector[sgm.perm_inds_[self.n_seeds_] - self.n_seeds_] += 1.0
+
+        prob_vector /= self.n_init
+
+        # Get the original vertices names in the B graph to make the nom list
+        b_inds = b_reord[ind2]
+
+        # Generate the nomination list. Note, the probability matrix does not
+        # include the seeds, so we must remove them from b_inds. Return a list
+        # sorted so it returns the vertex with the highest probability first.
+        nomination_list_ = list(zip(b_inds[self.n_seeds_ :], prob_vector))
+        nomination_list_.sort(key=lambda x: x[1], reverse=True)
```

Comment:
should be a way to do this with numpy functions so you don't have to cast and recast

Reply:
I thought this was more pretty than the np way, but I have changed it.